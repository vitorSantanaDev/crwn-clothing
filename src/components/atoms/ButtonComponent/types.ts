import { ReactNode } from 'react'

export enum ButtonTypeEnum {
  BUTTON = 'button',
  SUBMIT = 'submit',
  RESET = 'reset'
}

export enum ButtonTypeStyleEnum {
  DEFAULT = 'default',
  INVERTED = 'inverted',
  GOOGLE_SIGN_IN = 'google-sign-in'
}

export interface IButtonProps {
  children: ReactNode
  type?: ButtonTypeEnum
  buttonTypeStyle?: ButtonTypeStyleEnum
}
