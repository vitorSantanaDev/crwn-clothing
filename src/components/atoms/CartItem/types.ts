import { IProductCartItem } from 'interfaces'

export type ICartItemProps = IProductCartItem
