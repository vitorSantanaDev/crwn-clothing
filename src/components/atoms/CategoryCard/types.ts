export interface IViewProps {
  title: string
  action: string
  urlImageBackground: string
}
