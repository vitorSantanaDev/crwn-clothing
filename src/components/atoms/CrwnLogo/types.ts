export interface IViewProps {
  className?: string
}
