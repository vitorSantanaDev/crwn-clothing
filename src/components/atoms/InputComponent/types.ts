export type InputComponentProps = {
  name: string
  type: string
  placeholder: string
  errorComponent: string
}
