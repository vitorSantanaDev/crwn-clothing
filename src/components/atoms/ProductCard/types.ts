import { IProduct } from 'interfaces'

export type IProductCardProps = IProduct
