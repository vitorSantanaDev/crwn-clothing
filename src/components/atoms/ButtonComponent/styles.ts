import styled, { css } from 'styled-components'

import { ButtonTypeStyleEnum } from './types'

type ButtonProps = {
  buttonTypeStyle: ButtonTypeStyleEnum
}

export const ButtonElement = styled.button<ButtonProps>`
  min-width: 165px;
  width: auto;
  height: 50px;
  letter-spacing: 0.5px;
  line-height: 50px;
  padding: 0 35px 0 35px;
  font-size: 15px;
  background-color: black;
  color: white;
  text-transform: uppercase;
  font-weight: bolder;
  border: none;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: center;

  &:hover {
    background-color: white;
    color: black;
    border: 1px solid black;
  }
  &:disabled {
    cursor: not-allowed;

    &:hover {
      color: white;
      background-color: black;
    }
  }

  ${({ buttonTypeStyle }) =>
    buttonTypeStyle === ButtonTypeStyleEnum.GOOGLE_SIGN_IN &&
    css`
       {
        background-color: #4285f4;
        color: white;

        &:hover {
          background-color: #357ae8;
          border: none;
        }
      }
    `}

  ${({ buttonTypeStyle }) =>
    buttonTypeStyle === ButtonTypeStyleEnum.INVERTED &&
    css`
      background-color: white;
      color: black;
      border: 1px solid black;

      &:hover {
        background-color: black;
        color: white;
        border: none;
      }
    `}
`
