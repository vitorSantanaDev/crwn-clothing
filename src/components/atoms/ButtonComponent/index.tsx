import { ButtonTypeEnum, ButtonTypeStyleEnum, IButtonProps } from './types'
import * as S from './styles'

export default function ButtonComponent({
  children,
  type = ButtonTypeEnum.BUTTON,
  buttonTypeStyle = ButtonTypeStyleEnum.DEFAULT
}: IButtonProps) {
  return (
    <S.ButtonElement type={type} buttonTypeStyle={buttonTypeStyle}>
      {children}
    </S.ButtonElement>
  )
}
