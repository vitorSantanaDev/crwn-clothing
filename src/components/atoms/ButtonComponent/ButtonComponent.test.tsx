import { render, screen } from '@testing-library/react'

import { ButtonTypeStyleEnum } from './types'

import ButtonComponent from '.'
import userEvent from '@testing-library/user-event'

describe('<ButtonComponent />', () => {
  it('should render the button with default style', () => {
    render(
      <ButtonComponent buttonTypeStyle={ButtonTypeStyleEnum.DEFAULT}>
        button Text
      </ButtonComponent>
    )

    const button = screen.getByRole('button', { name: /button text/i })

    expect(button).toBeInTheDocument()
    expect(button).toHaveStyle({
      'background-color': 'black',
      color: 'white'
    })
  })

  it('should render the button with google sign in style', () => {
    render(
      <ButtonComponent buttonTypeStyle={ButtonTypeStyleEnum.GOOGLE_SIGN_IN}>
        button Text
      </ButtonComponent>
    )

    const button = screen.getByRole('button', { name: /button text/i })

    expect(button).toBeInTheDocument()
    expect(button).toHaveStyle({
      'background-color': '#4285f4',
      color: 'white'
    })
  })

  it('should render the button with inverted style', () => {
    render(
      <ButtonComponent buttonTypeStyle={ButtonTypeStyleEnum.INVERTED}>
        button Text
      </ButtonComponent>
    )

    const button = screen.getByRole('button', { name: /button text/i })

    expect(button).toBeInTheDocument()
    expect(button).toHaveStyle({
      'background-color': 'white',
      color: 'black',
      border: '1px solid black'
    })
  })

  it('should called the handleClick function when click in the button', () => {
    const handleClick = jest.fn()

    render(
      <ButtonComponent
        handleClick={handleClick}
        buttonTypeStyle={ButtonTypeStyleEnum.DEFAULT}
      >
        button Text
      </ButtonComponent>
    )

    const button = screen.getByRole('button', { name: /button text/i })

    expect(button).toBeInTheDocument()
    userEvent.click(button)
    expect(handleClick).toBeCalledTimes(1)
  })
})
