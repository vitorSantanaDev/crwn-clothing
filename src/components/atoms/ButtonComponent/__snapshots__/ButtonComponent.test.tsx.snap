// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonComponent /> should render the button with default style 1`] = `
<button
  class="sc-bczRLJ dkIoSW"
  type="button"
>
  button Text
</button>
`;

exports[`<ButtonComponent /> should render the button with google sign in style 1`] = `
<button
  class="sc-bczRLJ PrSTo"
  type="button"
>
  button Text
</button>
`;

exports[`<ButtonComponent /> should render the button with inverted style 1`] = `
<button
  class="sc-bczRLJ dvZaef"
  type="button"
>
  button Text
</button>
`;
